core/src/main/java/io/undertow/server/HttpServerExchange.java

```diff
@@ -953,6 +953,10 @@ public HttpServerExchange addExchangeCompleteListener(final ExchangeCompletionLi
         return this;
     }
 
+    public boolean hasListenersBeenInvoked() {
```

Comment:
Oh, just now I noticed this. :-( This method is public. I won't merge this PR for now, this will be reviewed further before we get to a decision.